catalog/controller/module/tawkto.php

```diff
@@ -130,4 +134,34 @@ private function matchPatterns($current_page, $pages)
 	{
 		return UrlPatternMatcher::match($current_page, $pages);
 	}
+
+	/**
+	 * Get visitor details
+	 *
+	 * @return array[string][string]|null
+	 */
+	private function getVisitor()
+	{
+		$logged_in = $this->customer->isLogged();
+		if ($logged_in) {
+			$data = array(
+					'name' => $logged_in?$this->customer->getFirstName().' '.$this->customer->getLastName():null,
```

Comment:
$logged_in is check above already, not necessary to check  $logged_in for name and email again.

 Moreover setting name to null or email to null would have their values as null.

 eg

 ```
 {"name":null}
 ```